tools/cbmc/proofs/CBMCStubLibrary/tasksStubs.c

```diff
@@ -0,0 +1,46 @@
+#include "FreeRTOS.h"
+#include "task.h"
+#include "tasksStubs.h"
+
+#ifndef TASK_STUB_COUNTER
+	#define TASK_STUB_COUNTER 0;
+#endif
+
+/* 5 is a magic number, but we need some number here as a default value.
+   This value is used to bound any loop depending on xTaskCheckForTimeOut
+   as a loop bound. */
+#ifndef TASK_STUB_COUNTER_LIMIT
+	#define TASK_STUB_COUNTER_LIMIT 5;
+#endif
+
+
+static BaseType_t xCounter = TASK_STUB_COUNTER;
+static BaseType_t xCounterLimit = TASK_STUB_COUNTER_LIMIT;
+
+BaseType_t xTaskGetSchedulerState( void )
+{
+	return xState;
```

Comment:
What about `return nondet_bool() ? pdTRUE : pdFALSE` instead? The scheduler is just supposed to return only one of those values, and it seems xState is not used anywhere else.

Reply:
Interesting, it is used like this ```xTaskGetSchedulerState() == taskSCHEDULER_SUSPENDED``` in queue.c. I will use xState in proof harnesses later.

Reply:
You are right, it is supposed to return 3 different values, and these depends on two different flags (`xSchedulerRunning` and `uxSchedulerSuspended`).